onnxruntime/python/session.py

```diff
@@ -303,3 +330,77 @@ def clear_binding_inputs(self):
 
     def clear_binding_outputs(self):
         self._iobinding.clear_binding_outputs()
+
+
+class OrtValue:
+    '''
+    This class provides APIs to construct and deal with OrtValues.
+    '''
+    def __init__(self, ortvalue):
+        if isinstance(ortvalue, C.OrtValue):
+            self._ortvalue = ortvalue
+        else:
+            # An end user won't hit this error
+            raise ValueError("`Provided ortvalue` needs to be of type `onnxruntime.capi.onnxruntime_pybind11_state.OrtValue`")
+
+    @staticmethod
+    def tensor_from_numpy(numpy_obj, device_type='cpu', device_id=0):
+        '''
+        Factory method to construct an OrtValue (which holds a Tensor) from a given Numpy object
+        A copy of the data in the Numpy object is held by the OrtValue
```

Comment:
We maintain (and manage the life-cycle of) a copy of the data in the numpy object - users get to specify where the memory will live - CPU (by default) or CUDA (main use-case)